ports/stm32/usbdev/class/src/usbd_cdc_msc_hid.c

```diff
@@ -462,45 +458,54 @@ int USBD_SelectMode(usbd_cdc_msc_hid_state_t *usbd, uint32_t mode, USBD_HID_Mode
     uint8_t num_itf = 0;
     switch (usbd->usbd_mode) {
         case USBD_MODE_MSC:
-            n += make_msc_desc(d + n);
-            num_itf = 1;
+            usbd->MSC_BOT_ClassData.interface = num_itf;
```

Comment:
Can you really use this variable `MSC_BOT_ClassData.interface` safely?  Note that the `USB_REQ_SET_INTERFACE` command will change this value... not sure what to but it might be better to use a separate (new) variable for this.

Reply:
Tbh this hasn't been thoroughly tested, last build I didn't have msc enabled.